src/router.rs

```diff
@@ -0,0 +1,86 @@
+#[derive(PartialEq, Clone)]
+pub enum MethodType {
+    POST,
+    GET,
+    PATCH,
+    DELETE,
+    CONNECT,
+    OPTIONS,
+    TRACE,
+    HEAD,
+}
+
+#[derive(Clone)]
+pub struct Endpoint {
+    method: MethodType,
+}
+
+#[derive(Default, Clone)]
+pub struct Route {
+    path: String,
+    endpoint: Endpoint,
+}
+
+#[derive(Default, Clone)]
+pub struct Router {
+    routes: Vec<Route>,
+}
+
+impl Default for Endpoint {
+    fn default() -> Self {
+        Endpoint {
+            method: MethodType::GET,
+        }
+    }
+}
+
+impl Router {
+    pub fn new() -> Self {
+        Router::default()
+    }
+
+    pub fn add_route(mut self, route: Route) -> Self {
```

Comment:
&mut self ?

Reply:
It's not a bit strange to return a `&mut Self`? Cause if I am adding a `&mut self` I have to return `&mut Self` if I want to chain the method? no?

Reply:
It seems to be the default behavior for https://github.com/colin-kiegel/rust-derive-builder#how-it-works , so i guess it s the preferred way ? I don't have a strong opinion on this either way tbh ^^